lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -4,109 +4,136 @@ module Hyacinth
       class Datacite < Abstract
-        IDENTIFIER_STATUS = { public: 'public',
-                              reserved: 'reserved',
-                              unavailable: 'unavailable' }.freeze
+        DOI_STATES = [:draft, :findable, :registered].freeze
+
+        # https://schema.datacite.org/meta/kernel-4.4/doc/DataCite-MetadataKernel_v4.4.pdf
+        REQUIRED_DATACITE_METADATA_PROPERTIES = [:creators,
+                                                 :publisher,
+                                                 :publication_year,
+                                                 :resource_type_general,
+                                                 :title,
+                                                 :url].freeze
+
+        # https://schema.datacite.org/meta/kernel-4.4/doc/DataCite-MetadataKernel_v4.4.pdf
+        SUPPORTED_DATACITE_METADATA_PROPERTIES = [:subject,
+                                                  :stuff].concat(REQUIRED_DATACITE_METADATA_PROPERTIES).freeze
+        attr_reader :rest_api
+
         def initialize(adapter_config = {})
           super(adapter_config)
         end
 
         # @param id [String]
-        # @return [Boolean] true if this adapter can handle this type of identifier
+        # @return [Boolean] true if this adapteppr can handle this type of identifier
         def handles?(id)
           id =~ /^10\.[A-Za-z0-9]+\/[A-Za-z0-9\-]+/
         end
 
-        # Generates a new persistent id, ensuring that nothing currently uses that identifier.
-        # @return [String] a new id
-        def mint
-          # metadata is blank when minting
-          # TODO: allow existing object data to be pushed on mint
-          metadata = {}
-          # target is blank when minting
-          # TODO: allow existing object data to be pushed on mint
-          target_url = nil
-          # status is reserved when minting
-          # TODO: allow provided target
-          identifier_status = IDENTIFIER_STATUS[:reserved]
-          # mint_identifier returns a Hyacinth::Ezid::Doi
-          ezid_doi_instance = api_session.mint_identifier(DATACITE[:shoulder][:doi], identifier_status, target_url, metadata)
-          # if above method returned nil, call to EZID API was unsuccessful. Return nil to indicate failure
-          # if unsuccessful because of timeout, log faiilure and return nil
-          if api_session.timed_out?
-            log(:error, "#mint_and_store_doi: EZID API call to mint_identifier timed-out.")
-            return nil
+        # metadata is not required when minting a draft DOI.
+        # metadata is required when minting a findable DOI. This metadata must include all the
+        # properties required by DataCite.
+        # It is also possible to specify a (non-existent) DOI when minting. However, this is not
+        # currently supported in the code.
+        def mint(hyacinth_metadata = nil, # Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite::Metadata
```

Comment:
we should discuss RDoc comment markup for expected parameter types

Reply:
+1

Reply:
@fcd1 I find this cheatsheet to be helpful when writing rdoc comments: https://devhints.io/rdoc

Reply:
Also, unrelated to rdoc, if this method changes the signature of `mint` from a parameter-less to one that takes several parameters, you'll also need to update the Abstract adapter (`lib/hyacinth/adapters/external_identifier_adapter/abstract.rb`) and Memory adapter (`lib/hyacinth/adapters/external_identifier_adapter/memory.rb`) signatures to match.

Also, it appears that you're only testing the parameter-less version of this method in your tests in `spec/hyacinth/adapters/external_identifier_adapter/datacite_spec.rb`, so you'll want to add tests for the parameterized version as well.

Reply:
Like my comment about the `#update` method, this method should take a digital object as a parameter rather than a custom `Datacite::Metadata` object -- but you can definitely convert the DigitalObject to `Datacite::Metadata` internally.  The idea behind the adapters methods is that the caller should, for the most part, just be able to say "adapter, take this object and extract whatever info you need from it."

Reply:
RDoc comments used where appropriate.
signatures for mint updated
test added for parameterized version of mint
parameter changed to digital object